File: csharp/darts/Darts.cs
using System;

public static class Darts
{
    private const int outerRadius = 10;
    private const int middleRadius = 5;
    private const int innerRadius = 1;
    public static int Score(double x, double y)
    {
        var squareSum = x * x + y * y;

        var points = squareSum switch
        {
            <= innerRadius * innerRadius => 10,
            <= middleRadius * middleRadius => 5,
            <= outerRadius * outerRadius => 1,
            _ => 0
        };

        return points;
    }
}

File: go/darts/darts.go
package darts

func Score(x, y float64) int {
	dist := x*x + y*y
	switch {
	case dist <= 1:
		return 10
	case dist <= 25:
		return 5
	case dist <= 100:
		return 1
	}
	return 0
}
